src/irmin-pack/layered.mli

```diff
@@ -0,0 +1,37 @@
+(*
+ * Copyright (c) 2013-2020 Thomas Gazagnaire <thomas@gazagnaire.org>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESIrmin. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+module Content_addressable
+    (H : Irmin.Hash.S)
+    (Index : Pack_index.S)
+    (Pack : Pack.S with type index = Index.t and type key = H.t) :
+  S.LAYERED_CONTENT_ADDRESSABLE_STORE
+    with type key = Pack.key
+     and type value = Pack.value
+     and type index = Pack.index
+     and module U = Pack
+     and module L = Pack
```

Comment:
I think having `U` and `L` in the code makes a bit more readable.

Reply:
Sure, I can see that. However, it's probably only true if the code doesn't apply `U` and `L` to the wrong stores, however, which is not enforced by the type system. Is it necessary to have them be equal, or is this done to avoid a functor parameter?

Reply:
I removed the module equality and give `U` and `L` as functor parameters, and the type system does not let me write `U.function t.lower` anymore. Thanks!

Reply:
Thanks, I prefer this :slightly_smiling_face: